### PR TITLE
add possibility to edit user profile

### PR DESCRIPTION
**Pull request by krizik**
added decorator for login, 404 route
need to work with validation and separately choice of inputs on edit page

### CODE REVIEW

**Review by yurapa — APPROVED** [review]

File: client/client_app/views/index.py [hunk 1]
```diff
@@ -46,6 +101,9 @@ def do_admin_login():
                 and logged_user.status != STATUS_BANNED):
             session['logged_in'] = logged_user.id
             session['name'] = logged_user.name
+            session['email'] = logged_user.email
+            session['phone'] = logged_user.phone
+            session['password'] = logged_user.password
```
Comment (serge30) [inline client/client_app/views/index.py:106]: Why do you need this in session?

File: client/client_app/views/index.py [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+"""
+Catch all 404 erorrs
+"""
+
+@app.errorhandler(404)
+def page_not_found(e):
+    return render_template('404.html'), 404
+  
+  
 class LoginForm(FlaskForm):
```
Comment (serge30) [inline client/client_app/views/index.py:41]: There is the same form class in the `client/client_app/models/login.py`

File: client/client_app/models/login.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+from flask_wtf import FlaskForm
+from wtforms import StringField, PasswordField
+from wtforms.validators import InputRequired, Email, Length
+
+
+class LoginForm(FlaskForm):
+    email = StringField(
+        'email',
+        validators=[InputRequired(), Email('Invalid Email')]
```
Comment (serge30) [inline client/client_app/models/login.py:13]: I'm not sure we need any other validation than `InputRequired` for login form.

File: client/client_app/models/login.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    )
+    password = PasswordField(
+        'password',
+        validators=[InputRequired(), Length(min=8)]
```
Comment (serge30) [inline client/client_app/models/login.py:17]: The same as above

File: client/client_app/views/index.py [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
 from client_app.models.restaurant import Restaurant
 from client_app.models.user import User
 
 
+"""
+Is Logged decorator. Put it in route you need to use only for logged user.
+Example:
+    @app.route('/home')
+    @is_logged
+"""
```
Comment (serge30) [inline client/client_app/views/index.py:21]: This is not docstring! Docstring goes after `def` statement.

File: client/client_app/views/index.py [hunk 6]
```diff
@@ -81,11 +140,29 @@ def register():
...
         return redirect(url_for('index'))
     return render_template('register.html', form=form)
 
 
+@app.route('/edit', methods=['GET', 'POST'])
+@is_logged
+def edit():
+    user_id = session['logged_in']
+    current_user = User.query.get(user_id)
+    form = edit_form.EditForm(request.form)
```
Comment (serge30) [inline client/client_app/views/index.py:153]: As this is a edit form, you need to populate form by current values stored in DB. Please check `obj` argument in [Form](https://wtforms.readthedocs.io/en/latest/forms.html#wtforms.form.Form) class.

Comment (krizik) [reply client/client_app/views/index.py:153]: I won't solve it to the end of this sprint. Task will be reopened and fixed in next one.

**Pull request merged**